relays/ethereum/src/substrate_sync_loop.rs

```diff
@@ -89,6 +91,16 @@ impl SourceClient<SubstrateHeadersSyncPipeline> for SubstrateHeadersSource {
 	type HeaderByHashFuture = Pin<Box<dyn Future<Output = (Self, Result<Header, Self::Error>)>>>;
 	type HeaderByNumberFuture = Pin<Box<dyn Future<Output = (Self, Result<Header, Self::Error>)>>>;
 	type HeaderExtraFuture = Ready<(Self, Result<(SubstrateHeaderId, ()), Self::Error>)>;
+	type HeaderCompletionFuture = Pin<
+		Box<
+			dyn Future<
+				Output = (
+					Self,
+					Result<(SubstrateHeaderId, Option<GrandpaJustification>), Self::Error>,
+				),
+			>,
+		>,
```

Comment:
That's `rustfmt`, right? Can we try to squeeze it into less lines? Maybe introducing:
```
type ClientOutput<X, T> = (X, Result<T, X::Error>);
```
would help here?